include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -842,52 +842,6 @@ class __future : private std::tuple<_Args...>
     }
 };
 
-// Invoke a callable and pass a compile-time integer based on a provided run-time integer.
-// The compile-time integer that will be provided to the callable is defined as the smallest
-// value in the integer_sequence not less than the run-time integer. For example:
-//
-//   __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, 2, 4, 8, 16>::__dispatch(f, 3);
-//
-// will call f<4>(), since 4 is the smallest value in the sequence not less than 3.
-//
-// If there are no values in the sequence less than the run-time integer, the last value in
-// the sequence will be used.
-//
-// Note that the integers provided in the integer_sequence must be monotonically increasing
-template <typename>
-class __static_monotonic_dispatcher;
```

Comment:
I think it is better to remove `__static_monotonic_dispatcher` together with its only use in `__parallel_copy_if`.

While it is a nice utility to avoid the verbose size-based runtime-to-template-instantiation dispatch with multiple `if` branches or `switch` statements, I think these explicitly verbose constructs are in a sense better - exactly because of "highlighting" the dispatch. This sort of dispatch is quite problematic due to multiple kernel instantiations, and it should be used consciously only where absolutely necessary. The nice wrapper both "hides" the ugly pattern and also adds an extra layer (if not two) to go through in case of refactoring or adjustments.